pubsub/google/cloud/pubsub/connection.py

```diff
@@ -419,7 +424,11 @@ def subscription_pull(self, subscription_path, return_immediately=False,
             'maxMessages': max_messages,
         }
         response = conn.api_request(method='POST', path=path, data=data)
-        return response.get('receivedMessages', ())
+        messages = response.get('receivedMessages', ())
+        for key, _ in enumerate(messages):
+            messages[key]['message']['data'] = base64.b64decode(
+                messages[key]['message']['data'])
```

Comment:
This identical to the code above except for the decode/encode: make a helper function that takes a decode/encode fn. as an argument.